python/ray/serve/backend_worker.py

```diff
@@ -68,10 +70,33 @@ async def __init__(self, backend_tag, replica_tag, init_args,
             if is_function:
                 _callable = backend
             else:
-                # This allows backends to define an async __init__ method
-                # (required for FastAPI backend definition).
-                _callable = backend.__new__(backend)
-                await sync_to_async(_callable.__init__)(*init_args)
+                # Retry with exponential backoff if ran into transient
+                # failures; Will throw and notify backend_state if
+                # constructor failed with no retries left
+                for current_try in range(REPLICA_CONSTRUCTOR_RETRY_COUNT + 1):
+                    # This allows backends to define an async __init__ method
+                    # (required for FastAPI backend definition).
+                    _callable = backend.__new__(backend)
+                    try:
+                        await sync_to_async(_callable.__init__)(*init_args)
+                    except Exception as e:
+                        logger.error(
+                            f"Exception while running deployment class "
+                            f"__init__: {e}")
+                        if current_try < REPLICA_CONSTRUCTOR_RETRY_COUNT:
+                            delay_secs = round(
+                                (2**current_try + random.uniform(0, 1)), 2)
+                            logger.info(
+                                f"Waiting for {delay_secs} secs to retry ..")
+                            await asyncio.sleep(delay_secs)
+                            continue
+                        else:
+                            logger.error(
+                                "Deployment constructor failed and ran out of "
+                                "retries. Aborting.")
+                            raise e
+                    break
```

Comment:
Can you move this break to directly after `await sync_to_async(_callable.__init__)(*init_args)`